src/common/utils/NebulaKeyUtils.h

```diff
@@ -251,6 +258,108 @@ class NebulaKeyUtils final {
         return rawKey.subpiece(0, rawKey.size() - sizeof(int64_t));
     }
 
+    // Only int and double are supported
+    static std::string boundVariant(nebula::cpp2::SupportedType type,
+                                    NebulaBoundValueType op,
+                                    const VariantType& v = 0L) {
+        switch (op) {
+            case NebulaBoundValueType::kMax : {
+                std::vector<unsigned char> bytes = {0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF};
+                return std::string(bytes.begin(), bytes.end());
+            }
+            case NebulaBoundValueType::kMin : {
+                std::string str;
+                str.reserve(sizeof(int64_t));
+                str.append(sizeof(int64_t), '\0');
+                return str;
+            }
+            case NebulaBoundValueType::kSubtraction : {
+                std::string str;
+                if (type == nebula::cpp2::SupportedType::INT) {
+                    str = encodeInt64(boost::get<int64_t>(v));
+                } else {
+                    str = encodeDouble(boost::get<double>(v));
+                }
+                std::vector<unsigned char> bytes(str.begin(), str.end());
+                for (size_t i = bytes.size(); i > 0; i--) {
```

Comment:
Is this logic correct? 
For example,  the bytes are "0x01  0x00"
As your code, the lower bound is "0x00 0x00" ??

Reply:
> Is this logic correct?
> For example, the bytes are "0x01 0x00"
> As your code, the lower bound is "0x00 0x00" ??

Good point， "0x01 0x00" - 1 should be "0x00 0xFF".